### ISSUE TITLE
1.4.7 Release breaks protractor-perf: browserName undefined

### ISSUE THREAD

**Issue by bcole** [axemclion/browser-perf#63]
With the latest release of browser-perf@1.4.7, protractor-perf now breaks on start:
```
TypeError: Cannot read property 'browserName' of undefined
    at Object.deepEquals (/Users/{project_path}/node_modules/protractor-perf/node_modules/browser-perf/lib/helpers.js:23:20)
    at AndroidTracingProbe.isEnabled (/Users/{project_path}/node_modules/protractor-perf/node_modules/browser-perf/lib/probes/AndroidTracingProbe.js:24:14)
    at AndroidTracingProbe.start (/Users/{project_path}/node_modules/protractor-perf/node_modules/browser-perf/lib/probes/AndroidTracingProbe.js:48:10)
    at /Users/{project_path}/node_modules/protractor-perf/node_modules/browser-perf/lib/probes/index.js:45:27
```

This is the commit that introduced `browserConfig` (which protractor-perf passes as undefined):
https://github.com/axemclion/browser-perf/commit/c1e6fe31190eb5ce866111be19c4587d5a0af010#diff-a0ccdc8e7313c6606025128061a22cc0
Since it is undefined, we cannot access .browserName.

The error is at `lib/probes/AndroidTracingProbe.js`, lines 46-48.

My suggestions are to either fix what the intention of this check was, or to roll back protractor-perf's package.json dependency on browser-perf to a hard 1.4.6.

**Comment by axemclion** [comment]
@bcole Thanks for catching the error. I think I will fix this error and release a new version.

**Comment by bcole** [comment]
Thanks for the quick reply!  Could you look into `helpers.js:26`?  That is still causing an undefined exception.  I think "global" itself is undefined, but you may have to debug a bit more.

New stack trace:
```
TypeError: Cannot read property 'browserName' of undefined
    at Object.deepEquals (/Users/{project_path}/node_modules/protractor-perf/node_modules/browser-perf/lib/helpers.js:26:18)
    at AndroidTracingProbe.isEnabled (/Users/{project_path}/node_modules/protractor-perf/node_modules/browser-perf/lib/probes/AndroidTracingProbe.js:24:14)
    at AndroidTracingProbe.start (/Users/{project_path}/node_modules/protractor-perf/node_modules/browser-perf/lib/probes/AndroidTracingProbe.js:48:10)
    at /Users/{project_path}/node_modules/protractor-perf/node_modules/browser-perf/lib/probes/index.js:45:27
```

**Comment by axemclion** [comment]
@bcole Can you send me your configuration, so that I can look at what exactly is failing ?

**Comment by bcole** [comment]
If global is not an object, 'return false' will not get executed.  The second condition would get executed.  I think your logic might be backward, perhaps you meant !== object.
https://github.com/axemclion/browser-perf/commit/8f11880b6e45233036abcc3fba492ca7802ebaef

I can't share my configuration, we have a private project.  It is a very standard usage of protractor-perf@0.2.1 though.  Do you have a sample protractor-perf project you can test with?

**Comment by axemclion** [comment]
@bcole I see what you are saying. Made some more changes, and wrote more tests for deepEquals - http://codepen.io/axemclion/pen/yJBGye?editors=0012

Can you add in your browserConfig and see if this still throws an exception ?

**Comment by bcole** [comment]
@axemclion I took a look at the callback within lifeCycle (AndroidTracingProbe.js:46) to see what 'browser' and 'browserConfig' are being set to, via protractor-perf.  protractor-perf seems to always set 'browserConfig' to undefined, and 'browser' to a large object.

Code (AndroidTracingProbe.js:46):
```
var lifeCycle = function(methodName) {
	return function(browser, browserConfig) {
		var me = this;
		if (me.isEnabled(browserConfig)) {
			return me[methodName](browser);
		}
	};
}
```

browserConfig:
```
undefined
```

browser:
```
EventEmitter {
  domain: null,
  _events: { status: [Function], command: [Function] },
  _eventsCount: 2,
  _maxListeners: undefined,
  sessionID: '161ac4bd-64a2-4053-a2d9-7e700ff20e70',
  configUrl: 
   Url {
     protocol: 'http:',
     slashes: true,
     auth: null,
     host: 'localhost:60000',
     port: 60000,
     hostname: 'localhost',
     hash: null,
     search: null,
     query: null,
     pathname: '/wd/hub',
     path: '/wd/hub',
     href: 'http://localhost:60000/wd/hub' },
  sauceRestRoot: 'https://saucelabs.com/rest/v1',
  noAuthConfigUrl: 
   Url {
     protocol: 'http:',
     slashes: true,
     host: 'localhost:60000',
     port: '60000',
     hostname: 'localhost',
     hash: null,
     search: null,
     query: null,
     pathname: '/wd/hub',
     path: '/wd/hub',
     href: 'http://localhost:60000/wd/hub' },
  defaultCapabilities: 
   { browserName: 'firefox',
     version: '',
     javascriptEnabled: true,
     platform: 'ANY' },
  _httpConfig: 
   { timeout: undefined,
     retries: 3,
     retryDelay: 15,
     baseUrl: undefined,
     proxy: undefined } }
```

--

Looks like you have a test case for `undefined`, so my case is now handled correctly.  Is it expected for browserConfig to be undefined from protractor-perf?

I manually copied your changes from that codepen to my installed version of helpers.js, and the issue seems to be resolved with that change.

**Comment by axemclion** [comment]
The browserConfig change was new - it is supposed to be the initial configuration that is passed. I may have to update protractor-perf to also send in this object, since it fixes #61. If browserConfig is empty in protractorPerf, it may have #61 issue.

**Comment by axemclion** [comment]
Closing this issue, since this is fixed in browser-perf.
